bundles/org.openhab.core/src/main/java/org/openhab/core/library/types/DateTimeType.java

```diff
@@ -248,4 +250,85 @@ private ZonedDateTime parse(String value) throws DateTimeParseException {
 
         return date;
     }
+
+    public boolean isToday() {
+        return sameDay(ZonedDateTime.now());
+    }
+
+    public boolean isTomorrow() {
+        return sameDay(ZonedDateTime.now().plusDays(1));
+    }
+
+    public boolean isYesterday() {
+        return sameDay(ZonedDateTime.now().minusDays(1));
+    }
+
+    public boolean sameDay(DateTimeType other) {
+        return sameDay(other.zonedDateTime);
+    }
+
+    public boolean sameDay(ZonedDateTime other) {
+        return zonedDateTime.truncatedTo(ChronoUnit.DAYS)
+                .isEqual(other.withZoneSameInstant(zonedDateTime.getZone()).truncatedTo(ChronoUnit.DAYS));
+    }
+
+    public DateTimeType toToday() {
+        return shiftDaysFromToday(0);
+    }
+
+    public DateTimeType toTomorrow() {
+        return shiftDaysFromToday(1);
+    }
+
+    public DateTimeType toYesterday() {
+        return shiftDaysFromToday(-1);
+    }
+
+    private DateTimeType shiftDaysFromToday(int days) {
+        ZonedDateTime now = ZonedDateTime.now().plusDays(days);
+        return new DateTimeType(zonedDateTime.withYear(now.getYear()).withMonth(now.getMonthValue())
+                .withDayOfMonth(now.getDayOfMonth()));
+    }
+
```

Comment:
This also has the same ambiguity issue related to differences in time zones between what's in `zonedDateTime` vs the system time zone.

```
jshell> var d1 = java.time.ZonedDateTime.parse("2024-01-01T00:00:00+11:00")
d1 ==> 2024-01-01T00:00+11:00

jshell> var now = java.time.ZonedDateTime.parse("2025-05-01T00:00:00-11:00")
now ==> 2025-05-01T00:00-11:00

jshell> d1.withYear(now.getYear()).withMonth(now.getMonthValue())
   ...>                 .withDayOfMonth(now.getDayOfMonth()).withZoneSameInstant(now.getZone())
$38 ==> 2025-04-30T02:00-11:00
```

Reply:
Made it relative to `zonedDateTime` zone's.